docs/public-api.asciidoc

```diff
@@ -0,0 +1,265 @@
+});
+----
+
+This API also supports `async` methods with the `Func<Task>` overloads.
+
+NOTE: The duration of the transaction will be the timespan between the first and the last line of the `async` lambda expression.
+
+Example:
+[source,csharp]
+----
+await Elastic.Apm.Agent.Api
+        .CaptureTransaction("TestTransaction", "TestType", async () =>
+{
+    //application code that is captured as a transaction
+    await Task.Delay(500); //sample async code
+});
+----
+
+NOTE: The <<convenient-capture-transaction>> method does not start any `Task` for you, so you have to call it with the `await` keyword if you use one of the overloads that except `Task` or `Task<T>`.
+
+
+//----------------------------
+[float]
+[[api-transaction]]
+=== Transaction API
+//----------------------------
+A transaction is the data captured by an agent representing an event occurring in a monitored service
+and groups multiple spans in a logical group.
+A transaction is the first <<api-span,`Span`>> of a service, and is also known under the term entry span.
+
+See <<api-current-transaction>> on how to get a reference of the current transaction.
+
+NOTE: Calling any of the transaction's methods after <<api-transaction-end>> has been called is illegal.
+You may only interact with transaction when you have control over its lifecycle.
+
+[float]
+[[api-transaction-create-span]]
+==== `Span StartSpan(string name, string type, string subType = null, string action = null)`
+Start and return a new custom span as a child of the given transaction.
+
+It is important to call <<api-span-end>> when the span has ended or to use the <<convenient-capture-span>> method.
+A best practice is to use the span in a try-catch-finally block.
+
+Example:
+
+[source,csharp]
+----
+ISpan span = transaction.StartSpan("Select FROM customer", Span.TYPE_DB);
+try
+{
+    span.Subtype = Span.SUBTYPE_MSSQL;
+    span.Action = Span.ACTION_QUERY;
+    //execute db query
+}
+catch(Exception e)
+{
+    span.CaptureException(e);
+    throw;
+}
+finally
+{
+    span.End();
+}
+----
+
+
+[float]
+[[api-transaction-end]]
+==== `void End()`
+Ends the transaction and schedules it to be reported to the APM Server.
+It is illegal to call any methods on a span instance which has already ended.
+This also includes this method and <<api-transaction-create-span>>.
+
+Example:
+
+[source,csharp]
+----
+transaction.End();
+----
+
+NOTE: If you use the <<convenient-capture-transaction>> method you must not call <<api-transaction-end>>.
+
+
+[float]
+[[api-transaction-capture-exception]]
+==== `void CaptureException(Exception e)`
+Captures an exception and reports it to the APM server.
+
+[float]
+[[api-transaction-capture-error]]
+==== `void CaptureError(string message, string culprit, StackFrame[] frames)`
+Captures a custom error and reports it to the APM server. This method is typically used when you want to report an error, but you don't have an `Exception` instance.
```

Comment:
I think you should add a blank line after:

> Captures a custom error and reports it to the APM server.

Reply:
done